doc/user-guide.md

```diff
@@ -400,6 +400,70 @@ func main() {
 
 After adding new import paths to your operator project, run `dep ensure` in the root of your project directory to fulfill these dependencies.
 
+## Leader election
+
+For improved reliability and quick failover it may be necessary to run multiple replicas of the operator with leader election, so that one leader instance handles the reconciliation while the other instances are inactive but ready to take over when the leader fails.
```

Comment:
As far as I know it's mostly the control plane controllers (like the scheduler) for whom high availability is an important requirement.
But I agree that improved reliability is not the dominant use case for using leader election and we probably shouldn't give that impression. I'll update this to point out that the focus is to avoid contention between multiple instances during rollouts.